Terminal.Gui/Input/Responder.cs

```diff
@@ -187,6 +192,7 @@ public void Dispose ()
 	{
 		// Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
 		Dispose (disposing: true);
+		Disposing?.Invoke (this, EventArgs.Empty);
```

Comment:
I think this line should be moved to after the `wasDisposed = true` line.

Reply:
Ah I only saw this now, in this case `Disposed` is the right name.

Reply:
Done.

Reply:
@tig we need to resolve this. I need to invoke the event before the `GC.SuppressFinalize` call. Then I need your finally opinion of the event name, `Disposing` or `Disposed`?

Reply:
> @tig we need to resolve this. I need to invoke the event before the `GC.SuppressFinalize` call. Then I need your finally opinion of the event name, `Disposing` or `Disposed`?

Explain why please.

Reply:
Although the event is invoked on the `Dispose` method it must be called before calling the `GC.SuppressFinalize` to avoid unit test failure. So it isn't already disposed, although the `View` already disposed others objects. So do you confirm that you want the event named as `Disposed`. If affirmative then it's already changed.

Reply:
> Although the event is invoked on the `Dispose` method it must be called before calling the `GC.SuppressFinalize` to avoid unit test failure. So it isn't already disposed, although the `View` already disposed others objects. So do you confirm that you want the event named as `Disposed`. If affirmative then it's already changed.

Are you sure the unit test is valid?

Reply:
Of course it's. Now it only handle one view as you asked. But if you think it isn't valid please say why you are think that.

Reply:
> Of course it's. Now it only handle one view as you asked. But if you think it isn't valid please say why you are think that.

Sorry, but I don't have time to dive deeply into this and understand it myself. I'd like you to explain to me what fails if the event is fired after `GC.SuppresFinalize`. With my limited understanding it should make any difference. 

I can't answer on what the event should be named without understanding this.

Reply:
Because when the event is invoked `container1` was already had disposed the view and the `container2` could not dispose all the `Responder.Instances`. See the error below. But this failure is arbitrary. Before I push the commit all the unit tests passed locally but failed on the server after pushed it.

```
  Failed Terminal.Gui.InputTests.ResponderTests.Disposed_Event_Notify_All_Subscribers_On_The_First_Container [2 ms]
  Error Message:
   Assert.Empty() Failure: Collection was not empty
Collection: [FrameView()(0,0,0,0), Frame(Margin)(0,0,0,0), Frame(Border)(0,0,0,0), Frame(Padding)(1,1,0,0), FrameView(Test)(0,0,0,0), ···]
  Stack Trace:
     at Terminal.Gui.InputTests.ResponderTests.Disposed_Event_Notify_All_Subscribers_On_The_First_Container() in D:\a\Terminal.Gui\Terminal.Gui\UnitTests\Input\ResponderTests.cs:line 238
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
  Failed Terminal.Gui.InputTests.ResponderTests.Responder_Not_Notifying_Dispose [< 1 ms]
  Error Message:
   Assert.Empty() Failure: Collection was not empty
Collection: [FrameView()(0,0,0,0), Frame(Margin)(0,0,0,0), Frame(Border)(0,0,0,0), Frame(Padding)(1,1,0,0), FrameView(Test)(0,0,0,0), ···]
  Stack Trace:
     at Terminal.Gui.InputTests.ResponderTests.Responder_Not_Notifying_Dispose() in D:\a\Terminal.Gui\Terminal.Gui\UnitTests\Input\ResponderTests.cs:line 168
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

Reply:
I still don't understand. 

However, I now think that the event should be `Disposing` and should be fired before `protected virtual void Dispose (bool disposing)` is called.

Reply:
The API doc should read:

`/// Event raised when <see cref="Dispose()"/> has been called to signal that this object is being disposed.`

Reply:
> I still don't understand.

Me either 😃 but I can't debug it because I don't get any fail locally. Some racing condition caused by the event being called after `GC.SuppressFinalize` which depend of the OS to do the cleanup from the garbage collector.